templates/BuildMyCharm/config.xml

```diff
@@ -51,7 +51,14 @@ charm build
 
 juju switch {{controller}}
 juju add-model test-$BUILD_NUMBER
-bundletester -e {{controller}}:test-$BUILD_NUMBER -vFt /tmp/builds/{{charmname}}
+
+if [ -z "{{referencebundle}}"]
+then
+    bundletester -e {{controller}}:test-$BUILD_NUMBER -vFt /tmp/builds/{{charmname}}
+else
+    juju deploy /tmp/builds/{{charmname}}
```

Comment:
This should ensure it deploys to the correct controller.  It should also get the service name from the bundle to ensure that the locally built charm is used.